front/app/modules/commercial/representativeness/admin/hooks/createRefDataSubscription.ts

```diff
@@ -0,0 +1,229 @@
+    userCustomFieldId
+  ).observable;
+
+  const subscription = observable.subscribe(handleRegFieldResponse(setters));
+
+  return subscription;
+};
+
+// Helpers
+const handleRegFieldResponse =
+  ({ setReferenceData, setIncludedUsers, setReferenceDataUploaded }: Setters) =>
+  (usersByRegField: IUsersByRegistrationField | NilOrError) => {
+    if (isNilOrError(usersByRegField)) {
+      setReferenceData(usersByRegField);
+      setIncludedUsers(usersByRegField);
+      return;
+    }
+
+    if (!usersByRegField.series.reference_population) {
+      setReferenceDataUploaded(false);
+      return;
+    }
+
+    setReferenceData(regFieldToReferenceData(usersByRegField));
+    setIncludedUsers(regFieldToIncludedUsers(usersByRegField));
+    setReferenceDataUploaded(true);
+  };
+
+export const regFieldToReferenceData = (
+  usersByField: IUsersByRegistrationField
+): RepresentativenessRowMultiloc[] => {
+  const { users, reference_population } = usersByField.series;
+
+  const optionIds = Object.keys(reference_population);
+  const includedUsers = syncKeys(users, optionIds);
+
+  const options = usersByField.options;
+
+  const actualPercentages = roundPercentages(Object.values(includedUsers), 1);
+  const referencePercentages = roundPercentages(
+    Object.values(reference_population),
+    1
+  );
+
+  const actualPercentagesObj = zipObject(optionIds, actualPercentages);
+  const referencePercentagesObj = zipObject(optionIds, referencePercentages);
+
+  const sortedOptionIds = [...optionIds].sort(
+    (a, b) => options[a].ordering - options[b].ordering
+  );
+
+  const data = sortedOptionIds.map((optionId) => ({
+    title_multiloc: options[optionId].title_multiloc,
+    actualPercentage: actualPercentagesObj[optionId],
+    referencePercentage: referencePercentagesObj[optionId],
+    actualNumber: includedUsers[optionId],
+    referenceNumber: reference_population[optionId],
+  }));
+
+  return data;
+};
+
+// Makes sure that users has the same keys as reference_population
+const syncKeys = (users: Record<string, number>, keys: string[]) => {
+  const fixedUsers = {};
+
+  keys.forEach((key) => {
+    fixedUsers[key] = users[key] ?? 0;
+  });
+
+  return fixedUsers;
+};
+
+export const regFieldToIncludedUsers = (
+  usersByField: IUsersByRegistrationField
+): IncludedUsers => {
+  const { users, reference_population } = usersByField.series;
+  const includedUsers = syncKeys(users, [
+    ...Object.keys(reference_population),
+    '_blank',
+  ]);
+
+  const known = Object.keys(includedUsers)
+    .filter((optionId) => optionId !== '_blank')
+    .reduce((acc, v) => users[v] + acc, 0);
+
+  const total = sum(Object.values(includedUsers));
+  const percentage = total === 0 ? 0 : roundPercentage(known, total);
+
+  return {
+    known,
+    total,
+    percentage,
+  };
+};
+
+export const ageFieldToReferenceData = (
+  { series: { user_counts, reference_population, bins } }: IUsersByAge,
+  locale: Locale
```

Comment:
It's a bit silly, but we need a `title_multiloc` as return value here, so I am creating a fake `multiloc` using the current locale

Reply:
yeah i think is the only case where a display text doesn't have any translation possible, so its fine